core/services/src/main/java/org/exoplatform/services/attachments/service/AttachmentServiceImpl.java

```diff
@@ -142,15 +156,19 @@ public void updateEntityAttachments(long userIdentityId,
     if (attachmentIds != null && !attachmentIds.isEmpty()) {
       for (String attachmentId : attachmentIds) {
         if (!existingAttachmentsIds.contains(attachmentId) && StringUtils.isNotEmpty(attachmentId)) {
-          linkAttachmentsToEntity(userIdentityId, entityId, entityType, Collections.singletonList(attachmentId));
+          try {
+            linkAttachmentToEntity(userIdentityId, entityId, entityType, attachmentId);
+          } catch (Exception e) {
+            throw new ObjectNotFoundException("Can't link attachment with id " + attachmentId + " to entity type " + entityType
```

Comment:
the thrown exception here is not always saying the truth IMO. (you throw ObjectNotFoundException for any type of exception, even, for example, session closed...)